pkg/blockchain/rpcLogStreamer.go

```diff
@@ -127,24 +127,33 @@ func (r *RpcLogStreamer) Start() {
 func (r *RpcLogStreamer) watchContract(watcher ContractConfig) {
 	fromBlock := watcher.FromBlock
 	logger := r.logger.With(zap.String("contractAddress", watcher.ContractAddress.Hex()))
-	startTime := time.Now()
 	defer close(watcher.EventChannel)
 
+	timer := time.NewTimer(watcher.maxDisconnectTime)
+	defer timer.Stop()
+
 	for {
 		select {
 		case <-r.ctx.Done():
 			logger.Debug("Stopping watcher")
 			return
+		case <-timer.C:
+			logger.Fatal(
```

Comment:
Just want to triple check.

We are confident that if the chain is working properly but there are no events coming in on any matching contracts, this won't kill our node?

Reply:
correct. The two cases that take down the node is 
- r.client.BlockNumber is never succeeding
- r.client.FilterLogs is never succeeding

Empty logs is not considered a failure